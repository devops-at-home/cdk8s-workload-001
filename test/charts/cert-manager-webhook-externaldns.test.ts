import { App } from 'cdk8s';
import {
    CertManagerWebhookExternalDns,
    CertManagerWebhookExternalDnsConfig,
    parseInputs,
} from '../../src/charts/cert-manager-webhook-externaldns.chart';

const chart = 'cert-mnager-webhook-external-dns';

describe(chart, () => {
    test('Snapshot test', () => {
        const app = new App();

        const manifest = new CertManagerWebhookExternalDns(app, chart, {
            namespace: 'cert-manager',
            labels: {
                chart,
            },
        });

        expect(manifest.toJson()).toMatchSnapshot();
    });
});

describe('Unit tests', () => {
    test('parseInputs function default values', () => {
        const response = parseInputs({});
        expect(response.domain).toStrictEqual('example.com');
    });

    test('parseInputs function with inputs', () => {
        const inputs: CertManagerWebhookExternalDnsConfig = {
            domain: 'domain.com',
        };

        const response = parseInputs(inputs);
        expect(response.domain).toStrictEqual(inputs.domain);
    });
});
