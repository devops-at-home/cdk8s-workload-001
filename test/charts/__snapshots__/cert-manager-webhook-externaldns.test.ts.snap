// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cert-manager-webhook-external-dns Snapshot test 1`] = `
[
  {
    "apiVersion": "v1",
    "kind": "ServiceAccount",
    "metadata": {
      "labels": {
        "chart": "externaldns-webhook",
      },
      "name": "cert-manager-webhook-external-d-webhook-serviceaccount-c8d64936",
      "namespace": "cert-manager",
    },
  },
  {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": {
      "labels": {
        "chart": "externaldns-webhook",
      },
      "name": "cert-manager-webhook-external-dns-service-c8579fb8",
      "namespace": "cert-manager",
    },
    "spec": {
      "ports": [
        {
          "name": "https",
          "port": 443,
          "protocol": "TCP",
        },
      ],
      "selector": {
        "chart": "externaldns-webhook",
      },
      "type": "ClusterIP",
    },
  },
  {
    "apiVersion": "apiregistration.k8s.io/v1",
    "kind": "APIService",
    "metadata": {
      "annotations": {
        "cert-manager.io/inject-ca-from": "cert-manager/externaldns-webhook-tls",
      },
      "labels": {
        "chart": "externaldns-webhook",
      },
      "name": "v1alpha1.example.com",
      "namespace": "cert-manager",
    },
    "spec": {
      "group": "example.com",
      "groupPriorityMinimum": 1000,
      "service": {
        "name": "cert-manager-webhook-external-dns-service-c8579fb8",
        "namespace": "cert-manager",
      },
      "version": "v1alpha1",
      "versionPriority": 15,
    },
  },
  {
    "apiVersion": "cert-manager.io/v1",
    "kind": "Issuer",
    "metadata": {
      "labels": {
        "chart": "externaldns-webhook",
      },
      "name": "cert-manager-webhook-external-issuer-webhook-selfsign-c839f5b1",
      "namespace": "cert-manager",
    },
    "spec": {
      "selfSigned": {},
    },
  },
  {
    "apiVersion": "cert-manager.io/v1",
    "kind": "Issuer",
    "metadata": {
      "labels": {
        "chart": "externaldns-webhook",
      },
      "name": "cert-manager-webhook-external-dns-issuer-webhook-ca-c8df8874",
      "namespace": "cert-manager",
    },
    "spec": {
      "ca": {
        "secretName": "externaldns-webhook-ca",
      },
    },
  },
  {
    "apiVersion": "cert-manager.io/v1",
    "kind": "Certificate",
    "metadata": {
      "labels": {
        "chart": "externaldns-webhook",
      },
      "name": "cert-manager-webhook-external-dns-webhook-ca-c84eb2b3",
      "namespace": "cert-manager",
    },
    "spec": {
      "commonName": "ca.externaldns-webhook.cert-manager",
      "duration": "43800h",
      "isCA": true,
      "issuerRef": {
        "name": "cert-manager-webhook-external-issuer-webhook-selfsign-c839f5b1",
      },
      "secretName": "externaldns-webhook-ca",
    },
  },
  {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": {
      "labels": {
        "chart": "externaldns-webhook",
      },
      "name": "cert-manager-webhook-external-dns-webhook-deployment-c8152d45",
      "namespace": "cert-manager",
    },
    "spec": {
      "replicas": 1,
      "selector": {
        "matchLabels": {
          "chart": "externaldns-webhook",
        },
      },
      "template": {
        "spec": {
          "containers": [
            {
              "args": [
                "--tls-cert-file=/tls/tls.crt",
                "--tls-private-key-file=/tls/tls.key",
              ],
              "env": [
                {
                  "name": "GROUP_NAME",
                  "value": "example.com",
                },
              ],
              "image": "absaoss/cert-manager-webhook-externaldns:dev",
              "livenessProbe": {
                "httpGet": {
                  "path": "/healthz",
                  "port": 443,
                  "scheme": "HTTPS",
                },
              },
              "name": "externaldns-webhook",
              "ports": [
                {
                  "containerPort": 443,
                  "name": "https",
                  "protocol": "TCP",
                },
              ],
              "readinessProbe": {
                "httpGet": {
                  "path": "/healthz",
                  "port": 443,
                  "scheme": "HTTPS",
                },
              },
              "volumeMounts": [
                {
                  "mountPath": "/tls",
                  "name": "certs",
                  "readOnly": true,
                },
              ],
            },
          ],
          "serviceAccountName": "cert-manager-webhook-external-d-webhook-serviceaccount-c8d64936",
          "volumes": [
            {
              "name": "certs",
              "secret": {
                "secretName": "externaldns-webhook-tls",
              },
            },
          ],
        },
      },
    },
  },
  {
    "apiVersion": "rbac.authorization.k8s.io/v1",
    "kind": "ClusterRole",
    "metadata": {
      "labels": {
        "chart": "externaldns-webhook",
      },
      "name": "cert-manager-webhook-external-dns-cluster-role-c8d22721",
      "namespace": "cert-manager",
    },
    "rules": [
      {
        "apiGroups": [
          "externaldns.k8s.io",
        ],
        "resources": [
          "dnsendpoints",
        ],
        "verbs": [
          "create",
          "list",
          "delete",
        ],
      },
      {
        "apiGroups": [
          "flowcontrol.apiserver.k8s.io",
        ],
        "resources": [
          "flowschemas",
          "prioritylevelconfigurations",
        ],
        "verbs": [
          "list",
          "watch",
        ],
      },
    ],
  },
  {
    "apiVersion": "rbac.authorization.k8s.io/v1",
    "kind": "RoleBinding",
    "metadata": {
      "labels": {
        "chart": "externaldns-webhook",
      },
      "name": "externaldns-webhook:webhook-authentication-reader",
      "namespace": "kube-system",
    },
    "roleRef": {
      "apiGroup": "rbac.authorization.k8s.io",
      "kind": "Role",
      "name": "extension-apiserver-authentication-reader",
    },
    "subjects": [
      {
        "apiGroup": "",
        "kind": "ServiceAccount",
        "name": "cert-manager-webhook-external-d-webhook-serviceaccount-c8d64936",
        "namespace": "cert-manager",
      },
    ],
  },
  {
    "apiVersion": "rbac.authorization.k8s.io/v1",
    "kind": "ClusterRoleBinding",
    "metadata": {
      "labels": {
        "chart": "externaldns-webhook",
      },
      "name": "externaldns-webhook:auth-delegator",
      "namespace": "cert-manager",
    },
    "roleRef": {
      "apiGroup": "rbac.authorization.k8s.io",
      "kind": "ClusterRole",
      "name": "system:auth-delegator",
    },
    "subjects": [
      {
        "apiGroup": "",
        "kind": "ServiceAccount",
        "name": "cert-manager-webhook-external-d-webhook-serviceaccount-c8d64936",
        "namespace": "cert-manager",
      },
    ],
  },
  {
    "apiVersion": "rbac.authorization.k8s.io/v1",
    "kind": "ClusterRoleBinding",
    "metadata": {
      "labels": {
        "chart": "externaldns-webhook",
      },
      "name": "externaldns-webhook:flowcontrol",
      "namespace": "cert-manager",
    },
    "roleRef": {
      "apiGroup": "rbac.authorization.k8s.io",
      "kind": "ClusterRole",
      "name": "cert-manager-webhook-external-dns-cluster-role-c8d22721",
    },
    "subjects": [
      {
        "apiGroup": "",
        "kind": "ServiceAccount",
        "name": "cert-manager-webhook-external-d-webhook-serviceaccount-c8d64936",
        "namespace": "cert-manager",
      },
    ],
  },
  {
    "apiVersion": "cert-manager.io/v1",
    "kind": "Certificate",
    "metadata": {
      "labels": {
        "chart": "externaldns-webhook",
      },
      "name": "cert-manager-webhook-external-dns-tls-c87a3343",
      "namespace": "cert-manager",
    },
    "spec": {
      "dnsNames": [
        "externaldns-webhook",
        "externaldns-webhook.cert-manager",
        "externaldns-webhook.cert-manager.svc",
        "externaldns-webhook.cert-manager.svc.cluster.local",
      ],
      "duration": "8760h",
      "issuerRef": {
        "name": "cert-manager-webhook-external-dns-issuer-webhook-ca-c8df8874",
      },
      "secretName": "externaldns-webhook-tls",
    },
  },
]
`;
